.github/workflows/python-ci.yml

```diff
@@ -31,12 +31,14 @@ jobs:
     - uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install python dependencies
+    - name: Build Python package and its dependencies
+      working-directory: ./python
       run: |
-        python -m pip install --upgrade pip wheel
-        pip install -r ./python/requirements.txt
-        pip install pyspark==${{ matrix.spark-version }}
+        python -m pip install --upgrade poetry
```

Comment:
I would recommend to use [action](https://github.com/snok/install-poetry/) instead.